client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpRequest.java

```diff
@@ -171,7 +172,6 @@ public HttpRequest withBody(byte[] body, String mimeContentType) {
      */
     public HttpRequest withBody(HttpRequestBody body) {
```

Comment:
Since this method doesn't update the content length header, it seems like this could be a dangerous method for a customer to call. I'd suggest making it private or at least package private.

Reply:
I kept tinkering with this idea, thinking about what RequestPolicy use case could exist for calling this method, about what we could do with the docs or parameters to indicate "hey! you might need to set the content length!"

Then I tried making it package-private and I wasn't able to access it in RestProxy.java. So it goes.